java-lib/src/main/java/com/wavefront/ingester/StringLineIngester.java

```diff
@@ -84,4 +84,12 @@ public static String joinPushData(List<String> pushData) {
     index.add(pushData.length());
     return index;
   }
+
+  public static int pushDataSize(String pushData) {
+    int length = StringUtils.countMatches(pushData, PUSH_DATA_DELIMETER);
```

Comment:
Strictly speaking, that should never happen as a delimiter sequence would be considered invalid characters. StringLineIngester.joinPushData and unjoinPushData are also operating under an assumption that PUSH_DATA_DELIMITER doesn't naturally occur in the payload.